sys/vm/vm_mmap.c

```diff
@@ -2080,6 +2083,17 @@ vm_mmap_object(vm_map_t map, vm_pointer_t *addr, vm_offset_t max_addr,
 		docow |= MAP_CHECK_EXCL;
 	if ((flags & MAP_GUARD) != 0)
 		docow |= MAP_CREATE_GUARD;
+	/*
+	 * If we're creating a new shared resevation with anonymous
+	 * backing and capability permissions, let the vm system know so
+	 * the entry can be backed and OBJ_SHARECAP set.
+	 *
+	 * XXX: there should be a permission check...
+	 */
+	if ((flags & (MAP_ANON | MAP_SHARED | MAP_RESERVATION_CREATE)) ==
+	    (MAP_ANON | MAP_SHARED | MAP_RESERVATION_CREATE) &&
+	    (maxprot & (VM_PROT_READ_CAP | VM_PROT_WRITE_CAP)) != 0)
+		docow |= MAP_SHARECAP;
```

Comment:
This doesn't cover the case of shm largepage mappings (they don't go through vm_mmap_object()). Is that intentional?

Reply:
This case only covers new `MAP_ANON|MAP_SHARED` mappings and is unrelated stories POSIX shm. It triggers a case in `vm_map_insert1` that forces the mapping to be backed in order to set `OBJ_HASCAP` and `OBJ_SHARECAP` on it. For POSIX shm this accomplished in `shm_alloc`.